### PR TITLE
docs: Add disclaimer about OIDC enabled clusters

### PR DESCRIPTION
**Pull request by luchillo17**
Signed-off-by: Carlos Esteban Lopez <lcarlosesteb@vmware.com>

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: docs/features/kubernetes/configuration.md [hunk 1]
```diff
@@ -159,7 +159,12 @@ auth:
         audience: ${AUTH_OKTA_AUDIENCE}
 ```
 
-The following values are supported out-of-the-box by the frontend: `google`, `microsoft`, `okta`, `onelogin`.
+The following values are supported out-of-the-box by the frontend: `google`, `microsoft`,
+`okta`, `onelogin`.
+
+Take note that `oidcTokenProvider` is just the issuer for the token, you can use any
+of these with an OIDC enabled cluster, like using `microsoft` as the issuer for a GKE
+cluster.
```
Comment (jamieklassen) [inline docs/features/kubernetes/configuration.md:167]: GKE clusters actually don't use the "OpenID Connect Tokens" auth strategy by default -- they use access tokens, so it's probably "Webhook Token Authentication" under the covers. You might need to use a feature called ["Identity Service for GKE"](https://cloud.google.com/kubernetes-engine/docs/how-to/oidc) in order for the actual "OpenID Connect Token" strategy to be used.

EKS clusters can be configured with an OIDC provider, which does actually use the "OpenID Connect Tokens" auth strategy under the covers. Maybe that's a better example? At least it doesn't require the use of a special named feature like "Identity Service"...

Comment (luchillo17) [reply docs/features/kubernetes/configuration.md:167]: Yes, will change

**Review by jamieklassen — APPROVED** [review]
Table looks good!
Screen Shot 2022-09-21 at 10 08 26 PM

**Review by freben — APPROVED** [review]
Thanks!

**Pull request merged**